src/basic/tmpfile-util.c

```diff
@@ -67,7 +67,7 @@ int fopen_temporary(const char *path, FILE **ret_f, char **ret_temp_path) {
 
 /* This is much like mkostemp() but is subject to umask(). */
 int mkostemp_safe(char *pattern) {
-        int fd;
+        int fd = 0;
```

Comment:
A workaround is OK, but please add tiny comment that this is for gcc. Also, 0 is a valid fd number, so please make this -1 instead.

Reply:
done